homeassistant/components/ecovacs_mqtt/manifest.json

```diff
@@ -0,0 +1,11 @@
+{
+  "domain": "ecovacs_mqtt",
+  "name": "Ecovacs MQTT",
```

Comment:
No, they need to log in with username and password on the ecovacs server (which is the mqtt broker and rest to mqtt converter).
Only the bot speaks MQTT instead of XMPP (current integration, see comment above).

Do you have a better name for it?

Reply:
Do users use the same credentials to log in to both the old and the new integration?

Reply:
Yes
The problem is that ecovacs has no public documentation on their API and I have no bot with the old protocol to test it. As the `ecovacs` integration has some issues and the code was not updated recently, I don't can add support for it in my library. Also the effort it not worth it as soon or later XMPP will be not used anymore. Ecovacs doesn't sell vacuum that supports it since 8-10 years (Not sure when they stopped it)

Reply:
I think it will be a better user experience to instead extend the old integration with the new protocol and library. Ie ask users to log in, then determine which devices exist and which protocol should be used, and then select old code or new code.

We can import the old YAML config to the config flow.

Alternatively we only support the old devices via the import flow and only new devices via the user step, until someone can test an old device and wants to add support for a user step for that.

@mib1185 do you have an old device and can help with testing?